### ISSUE TITLE
MPC-HC freezes when attempting to play surveillance camera stream

### ISSUE THREAD

**Issue by Perplexer1** [clsid2/mpc-hc#2157]
I've been using the original MPC-HC versions (with the last that I used being 1.7.7) to play my surveillance camera stream, which worked fine. Stream URL which I pasted into "File -> Open File" was in the form of:
http://192.168.0.10/videostream.flv?user=xxx&pwd=yyy

When playback starts, a right-click on the video -> Properties -> Details shows:
"Video: MPEG4 Video (H264) 1280x720 35.833fps [V: h264 main L4.0, yuv420p, 1280x720]"

Now I downloaded the "next generation" version of MPC-HC (2.0.0) and tried to open the same URL/stream. The status bar at the bottom says "Opening..." but nothing happens. The MPC-HC window shows the default logo and playback controls are greyed out. Drop-down menus do unroll but trying to enter Options or trying to Close the program via the File menu will freeze the GUI and a kill through Task Manager will be required.

If I play the stream in VLC and record a few seconds of it to the hard drive, the resulting *.mp4 file will play normally in MPC-HC 2.0.0.

I'm not sure if this is a bug or if something in MPC-HC was intentionally changed which breaks this stream playback functionality.

**Comment by clsid2** [comment]
Must be an issue with the internal codecs. There is not really anything I can do about it without access to a failing stream.

Try some older 1.9.x versions.

You can also try unchecking HTTP in the internal filter settings.

**Comment by Perplexer1** [comment]
I spent some time and checked all the older versions I could find. The "break" actually occurs between the earlier versions:

MPC-HC.1.7.10 (LAV 0.66.0.0) (stream works)
MPC-HC.1.7.10.5 (LAV 0.66.0.39-git) (stream doesn't work anymore in this and later versions)

I tried copying the "LAVFilters" folder from the 1.7.10 version into the 1.7.10.5 version and then 1.7.10.5 also worked (was able to open the stream). So maybe the change in the LAV version from 0.66.0.0 to 0.6.6.0.39 broke it? Would that be possible based on the described behavior? 

Unchecking "HTTP(S)" from "Internal Filters -> Source Filters" list did not make a difference.

**Comment by clsid2** [comment]
Yes, a change in LAV Filters must be the cause. But I don't know what, so I can not fix it. Also, those versions are almost 8 years old, so it must be something very specific to your stream, given that there is no general known issue.

Have you tried https instead of http?

**Comment by Perplexer1** [comment]
The cameras are at least 5 years old models and don't support https. I don't have the same problems with the newer cameras so I guess I agree with you that it's probably not worth pursuing a solution. I would also probably also not find any interest from any other developer to fix this even if I provided the stream. I'll just have the old version on the computer that I can use for monitoring the stream as long as I have these particular cameras in use.

So this can then be closed. Thank you for prompt replies and clarification.

**Comment by adipose** [comment]
> The cameras are at least 5 years old models and don't support https. I don't have the same problems with the newer cameras so I guess I agree with you that it's probably not worth pursuing a solution. I would also probably also not find any interest from any other developer to fix this even if I provided the stream. I'll just have the old version on the computer that I can use for monitoring the stream as long as I have these particular cameras in use.
> 
> 
> 
> So this can then be closed. Thank you for prompt replies and clarification.

What camera model was it?

**Comment by Perplexer1** [comment]
The two camera streams that I noticed this with are from "Maginon IPC-25 HDC" and "Maginon IPC-40 C". As far as I know these are just two models among many that are based on the same chinese base model. Other camera brands / models that are essentially identical are "Foscam FI8908W", "StorageOptions SON-IPC1", "Rollei SafetyCam-20 HD", and many others.

**Comment by adipose** [comment]
mjpeg
640x480 or 320x240
30fps or 15fps

I wonder if you tried the same parameters with:

https://www.codeproject.com/Articles/371955/Motion-JPEG-Streaming-Server

Can you replicate the issue?

**Comment by Perplexer1** [comment]
MJPEG is the secondary stream from these cameras. It doesn't play in MPC-HC versions higher than 1.7.10 either. It just shows one frame and stays frozen on it while saying "Playing". But at least the program doesn't completely freeze requiring a kill through the task manager. I did manage to get the MJPEG stream to play if I changed output from "Enhanced Video Renderer (custom presenter)" to "Old Video Renderer". Then the stream actually played.

But the main camera stream (MPEG4 / H264) crashes the program if using "Enhanced Video Rendered (custom presenter" or "Old Video Renderer", or any other.

I never played around with "Motion JPEG Streaming Server" since I didn't really need to so far.
